### ISSUE TITLE
How to init matomo _after_ user has accepted terms?

### ISSUE THREAD

**Issue by tarlepp** [Arnaud73/ngx-matomo#91]
We've an issue because we don't want to initialize Matomo, before user has accepted the terms of service. How would we do this?

For now it looks like that init is causing that matomo is loaded _before_ user has accepted this.

**Comment by Arnaud73** [comment]
I would start by adding `requireConsent: true` to the MatomoConfig and then, once your user has accepted the terms of services, I would call `this.matomoTracker.setConsentGiven()` which will everything to your Matomo instance.

**Comment by tarlepp** [comment]
Actually that didn't do the trick for me - I needed to track that state myself to get all working as expected.

**Comment by Arnaud73** [comment]
If you'd like to check the "standard" behavior, then:

1. Clone this repo
2. Build ngx-matomo : `npm run build:ngx-matomo:prod`
3. Run the demo app : `npm run test`

You'll see that in this app, no tracking occurs before user consent is provided.

If your use case required something different, I'm willing to have a look at it!

**Comment by tarlepp** [comment]
Yeah, the main reason is that we need to use this plugin - https://plugins.matomo.org/AjaxOptOut - and if we do that `your.piwik/index.php?module=API&method=AjaxOptOut.isTracked` request tracking is started.

So I just resolved this by storing this tracking status to local storage and if it's accepted then I do that query otherwise not - simple solution after some testing.
